src/utils/GovNotifyEvent.ts

```diff
@@ -1,22 +1,58 @@
+import { ExtSessionEvent, SessionEvent } from "../models/SessionEvent";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { AppError } from "./AppError";
+import { Constants } from "./Constants";
+import { HttpCodesEnum } from "./HttpCodesEnum";
+import { Logger } from "@aws-lambda-powertools/logger";
+
 export interface GovNotifyEvent {
 	"Message": {
 		"userId": string;
 		"emailAddress": string;
 		"firstName": string;
 		"lastName": string;
 		"messageType": string;
+		"documentType"?: string;
+		"documentExpiryDate"?: string;
+		"poAddress"?: string;
+		"poVisitDate"?: string;
+		"poVisitTime"?: string;
 	};
 }
 
-export const buildGovNotifyEventFields = (userId: string, email: string, firstName: string, lastName: string): GovNotifyEvent => {
-
-	return {
-		Message : {
-			userId,
-			emailAddress: email,
-			firstName,
-			lastName,
-			messageType: "email",
-		},
-	};
+export const buildGovNotifyEventFields = (nameParts: { givenNames: string[]; familyNames: string[] }, sessionEvent: ExtSessionEvent | SessionEvent, emailType: string, logger: Logger ): GovNotifyEvent => {
+	
+	switch (emailType) {					
+		case Constants.OLD_EMAIL:
+			return {
+				Message : {
+					userId: sessionEvent.userId,
+					emailAddress: sessionEvent.userEmail,
+					firstName: nameParts.givenNames[0],
+					lastName: nameParts.familyNames[0],
+					messageType: "oldEmail",
+				},
+			};
+		case Constants.NEW_EMAIL:{
+			const newSessionEvent: ExtSessionEvent = new ExtSessionEvent(sessionEvent);
+			return {
+				Message : {
+					userId: newSessionEvent.userId,
+					emailAddress: newSessionEvent.userEmail,
+					firstName: nameParts.givenNames[0],
+					lastName: nameParts.familyNames[0],
+					documentType: newSessionEvent.documentType,
+					poAddress: newSessionEvent.postOfficeInfo[0].address + " " + newSessionEvent.postOfficeInfo[0].post_code,
+					poVisitDate: newSessionEvent.postOfficeVisitDetails[0].post_office_date_of_visit,
+					poVisitTime: newSessionEvent.postOfficeVisitDetails[0].post_office_time_of_visit,
+					documentExpiryDate: newSessionEvent.documentExpiryDate,
```

Comment:
Can I just check where we're going to format this date?

Reply:
at GovNotify lambda when we sent out the email

Reply:
Ok, is that lambda currently/going to be responsible for any other field formatting? If it's only this field it feels like it might make more sense to do it when we validate and save the data?

Reply:
we need to format the time to remove space per Mark's latest comment, so i would rather like to handle any formattings there.